src/locales.cc

```diff
@@ -0,0 +1,253 @@
+/*---------------------------------------------------------------------------*\
+                          ____  _ _ __ _ __  ___ _ _
+                         |_ / || | '_ \ '_ \/ -_) '_|
+                         /__|\_, | .__/ .__/\___|_|
+                             |__/|_|  |_|
+\*---------------------------------------------------------------------------*/
+
+#include <iostream>
+#include <fstream>
+#include <iterator>
+#include <list>
+
+#include <zypp/ZYpp.h>
+#include <zypp/base/Logger.h>
+#include <zypp/base/IOStream.h>
+#include <zypp/base/String.h>
+#include <zypp/base/Flags.h>
+#include <zypp/ui/Selectable.h>
+#include <zypp/base/Regex.h>
+#include <zypp/sat/LocaleSupport.h>
+
+#include "output/Out.h"
+#include "main.h"
+#include "getopt.h"
+#include "Table.h"
+#include "utils/messages.h"
+#include "utils/misc.h" // for xml_encode
+#include "locales.h"
+#include "solve-commit.h"
+
+using namespace std;
+using namespace boost;
+using namespace zypp;
+
+extern ZYpp::Ptr God;
+
+
+static std::string isRequestedLocale( zypp::Locale lang )
+{
+    if ( God->pool().isRequestedLocale( lang ) )
+      return _("Yes");
+    else
+      return _("No");
+}
+
+static void printLocaleList( Zypper & zypper, const zypp::LocaleSet &locales )
+{
+  Table tbl;
+  // header
+  TableHeader th;
+
+  // translators: header of table column - the language code, e.g. en_US
+  th << _("Code");
+  // translators: header of table column - the language, e.g. English (United States)
+  th << _("Language");
+  // translators: header of table column - is the language supported? (Yes/No)
+  th << _("Supported");
+
+  tbl << th;
+
+  for_( it, locales.begin(), locales.end() )
+  {
+    TableRow tr(3);
+
+    tr << (*it).code();
+    tr << (*it).name();
+    tr << isRequestedLocale(*it);
+    tbl << tr;
+  }
+
+  tbl.sort(1);
+
+  cout << tbl;
+}
+
+static void printXmlLocaleList( Zypper & zypper, const zypp::LocaleSet &locales )
+{
+
+}
+
+static void printLocalePackages( Zypper & zypper, const zypp::sat::LocaleSupport & myLocale )
+{
+  Table tbl;
+  TableHeader th;
+
+  // translators: header of table column - S is for 'Status' (package installed or not)
+  th << _("S");
+  // translators: header of table column - the name of the package
+  th << _("Name");
+  // translators: header of table column - the package summary
+  th << _("Description");
+
+  tbl << th;
+
+  for_( it, myLocale.selectableBegin(), myLocale.selectableEnd() )
+  {
+    TableRow tr(3);
+    zypp::ui::Status status = (*it)->status();
+
+    if ( status == zypp::ui::S_KeepInstalled || status == zypp::ui::S_Protected )
+      tr << "i";
+    else
+      tr << " ";
+    tr << (*it)->name();
+    tr << (*it)->theObj()->summary();
+
+    tbl << tr;
+  }
+
+  tbl.sort(1);
+
+  cout << tbl;
+}
+
+static zypp::LocaleSet relevantLocales( vector<string> localeArgs, bool glob )
```

Comment:
Have to have a closer look, but regex for matching locales like `de` or `de_DE`?

For a strict match I'd use Locale operator==, for a relaxed match, Locale. language() operator== if Locale.country() is empty.

    'de_DE' relaxed match -> 'de_DE' 
    'de' relaxed match -> 'de', 'de_DE', `de_*`

Reply:
Do you still think we need to change that?

Reply:
Done